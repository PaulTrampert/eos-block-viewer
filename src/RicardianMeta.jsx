import React from 'react';
import PropTypes from 'prop-types';

const RicardianMeta = ({
  title,
  icon,
  summary
}) => {
  return (
    <div>
      {
        title &&
        <h1>
          {icon && <img src={icon} style={{height: 50, width: 50}} alt={icon}/>}
          {title}
        </h1>
      }
      {
        summary &&
        <h2>
          {summary}
        </h2>
      }
    </div>
  );
}

RicardianMeta.propTypes = {
  title: PropTypes.string,
  icon: PropTypes.string,
  summary: PropTypes.string
}

export default RicardianMeta;