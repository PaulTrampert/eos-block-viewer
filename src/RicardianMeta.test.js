import React from 'react';
import {shallow} from 'enzyme';
import RicardianMeta from './RicardianMeta';

describe('RicardianMeta', () => {
  let subject;

  describe('render', () => {
    beforeEach(() => {
      subject = shallow(<RicardianMeta title="some title" icon="some icon" summary="the summary" />);
    });

    describe('when title is not set', () => {
      beforeEach(() => {
        subject.setProps({title: undefined});
      });
      it("doesn't render the header", () => {
        expect(subject).toMatchSnapshot();
      });
    });
    describe('when title is set', () => {
      describe('when icon is not set', () => {
        beforeEach(() => {
          subject.setProps({icon: undefined});
        });

        it('renders the header with no icon', () => {
          expect(subject).toMatchSnapshot();
        });
      });
      describe('when icon is set', () => {
        it('renders the header with an icon', () => {
          expect(subject).toMatchSnapshot();
        });
      });
    });

    describe('when summary is not set', () => {
      beforeEach(() => {
        subject.setProps({summary: undefined});
      });

      it('renders without a summary', () => {
        expect(subject).toMatchSnapshot();
      });
    });

    describe('when summary is set', () => {
      it('renders with a summary', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });
});