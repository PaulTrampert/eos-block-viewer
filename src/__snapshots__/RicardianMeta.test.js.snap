// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RicardianMeta render when summary is not set renders without a summary 1`] = `
<div>
  <h1>
    <img
      alt="some icon"
      src="some icon"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    some title
  </h1>
</div>
`;

exports[`RicardianMeta render when summary is set renders with a summary 1`] = `
<div>
  <h1>
    <img
      alt="some icon"
      src="some icon"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    some title
  </h1>
  <h2>
    the summary
  </h2>
</div>
`;

exports[`RicardianMeta render when title is not set doesn't render the header 1`] = `
<div>
  <h2>
    the summary
  </h2>
</div>
`;

exports[`RicardianMeta render when title is set when icon is not set renders the header with no icon 1`] = `
<div>
  <h1>
    some title
  </h1>
  <h2>
    the summary
  </h2>
</div>
`;

exports[`RicardianMeta render when title is set when icon is set renders the header with an icon 1`] = `
<div>
  <h1>
    <img
      alt="some icon"
      src="some icon"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    some title
  </h1>
  <h2>
    the summary
  </h2>
</div>
`;
